### ISSUE TITLE
Investigate asynchronous rpc request for forks

### ISSUE THREAD

**Issue by cptartur** [foundry-rs/starknet-foundry#792]
Blocked by #663 
It should be a shared state between different tests, which relies on some workers already having fetched some data.
It's an optimization and is not high prio

**Comment by Arcticae** [comment]
Currently blocked because async execution architecture is not stable

**Comment by Arcticae** [comment]
Duplicate of #901
